### ISSUE TITLE
False positive from `Lint/UselessAssignment` with module variable

### ISSUE THREAD

**Issue by marcandre** [rubocop-hq/rubocop#6405]
The following code produces an error "Lint/UselessAssignment: Useless assignment to variable - x" when it shouldn't.

```
x = Thread.current['x'] || Object
module x::Foo
end
```

```
$ [bundle exec] rubocop -V
0.59.2 (using Parser 2.5.1.2, running on ruby 2.5.1 x86_64-darwin15)
```
